ZFitter/script/functions.sh

```diff
@@ -110,37 +84,16 @@ mkSmearerCatData(){
     if [ ! -e "$2/smearerCat_`basename $1 .dat`_d1-`basename $configFile .dat`.root" ];then
 	echo "[STATUS] Creating smearerCat for data: `basename $configFile .dat` `basename $1 .dat`"
 	./bin/ZFitter.exe -f $3 --regionsFile=$1  \
-	    --saveRootMacro  --addBranch=smearerCat_d $4 || exit 1
+	    --saveRootMacro  --addBranch=smearerCat_d $4 #|| exit 1 -->FIX ME (dummy  seg faults)
 	mv tmp/smearerCat_`basename $1 .dat`_d*-`basename $configFile .dat`.root $2/ || exit 1
     fi
-# Not working for me (Giuseppe)
-#    cat $3 \
-#	| sed "/selected/ ! d; /selected/{ s|^\(d[1-9]\)\tselected.*|\1\tsmearerCat_`basename $1 .dat`\t$2/smearerCat_`basename $1 .dat`_\1-`basename $3 .dat`.root|}" | sort | uniq |grep smearerCat |grep '^d'   >> $3.tmp
-#    cat $3.tmp $3 | sort | uniq -d > $3.tmp2
-#    for line in `cat $3.tmp2 | sed 's|[ ]+|\t|g' | cut  -f 3`;
-#      do
-#      sed -i "\#$line# d" $3.tmp
-#    done
-#    cat $3.tmp
-#    echo 
-#    echo
-#    echo
-#    echo "file of mkSmearerCatData is" $3
-#    cat $3.tmp >> $3
-#    rm $3.tmp $3.tmp2
-
-##adapted from mkSmearerSignal -->This works
-#Once cat files exist, just write them in the bare configfile
+
+    #Once cat files exist, just write them in the bare configfile
+    #TO-DO -> be sure that the writing is unique to avoid stupid crashes
     for tag in `grep "^d" $3 | grep selected | awk -F" " ' { print $1 } '`
```

Comment:
Ok, thanks. The fix here would be "if the category file is already written in the validation file, do not write it again".